### ISSUE TITLE
``bad-format-character`` false negative on ``.format()`` and f-strings

### ISSUE THREAD

**Issue by DudeNr33** [PyCQA/pylint#6085]
### Bug description

# pylint: disable=consider-using-f-string, missing-module-docstring
PI = 3.1415
print("Pi is %.2v" % PI)
print("Pi is {:.2v}".format(PI))
print(f"Pi is {PI:.2v}")

### Command used

```shell
pylint a.py
```

### Pylint output

```shell
************* Module a
a.py:3:6: E1300: Unsupported format character 'v' (0x76) at index 9 (bad-format-character)
```

### Expected behavior

``bad-format-character`` is currently only emitted for the old-style string formatting.

It should also cover new style formatting and f-strings.

### Pylint version

```shell
pylint 2.14.0-dev0
astroid 2.11.2
Python 3.9.6 (default, Jul 28 2021, 21:15:06) 
[Clang 12.0.5 (clang-1205.0.22.9)]
```

### OS / Environment

macOS Big Sur Version 11.6

**Comment by DanielNoord** [comment]
For anybody taking this on, after #6086 has landed we will need to update the documentation for this as well.
